src/main/java/org/takes/facets/auth/PsBasic.java

```diff
@@ -0,0 +1,175 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.facets.auth;
+
+import java.io.IOException;
+import javax.xml.bind.DatatypeConverter;
+import lombok.EqualsAndHashCode;
+import org.takes.Request;
+import org.takes.Response;
+import org.takes.misc.Opt;
+import org.takes.rq.RqHeaders;
+
+/**
+ * Pass that checks the user according RFC-2617.
+ *
+ * <p>The class is immutable and thread-safe.
+ *
+ * @author Endrigo Antonini (teamed@endrigo.com.br)
+ * @version $Id$
+ * @since 0.20
+ */
+@EqualsAndHashCode(of = { "entry" })
+public final class PsBasic implements Pass {
+
+    /**
+     * Authorization response HTTP head.
+     */
+    private static final String AUTH_HEAD = "Basic";
+
+    /**
+     * Entry to validate user information.
+     */
+    private final transient PsBasic.Entry entry;
+
+    /**
+     * Ctor.
+     * @param basic Entry
+     */
+    public PsBasic(final PsBasic.Entry basic) {
+        this.entry = basic;
+    }
+
+    @Override
+    public Opt<Identity> enter(final Request request) throws IOException {
+        Opt<Identity> identity = new Opt.Empty<Identity>();
+        try {
+            final String decoded = new String(
+                DatatypeConverter.parseBase64Binary(
+                    new RqHeaders.Smart(
+                        new RqHeaders.Base(request)
+                    ).single("authorization").split(AUTH_HEAD)[1]
+                )
+            ).trim();
+            final String user = decoded.split(":")[0];
+            if (this.entry.check(
+                user,
+                decoded.substring(user.length() + 1)
+            ).has()) {
+                identity = new Opt.Single<Identity>(
+                    new Identity.Simple(
+                        String.format("urn:basic:%s", user)
+                    )
+                );
+            }
+        } catch (final IOException ex) {
+            identity = new Opt.Empty<Identity>();
+        }
+        return identity;
+    }
+
+    @Override
+    public Response exit(final Response response, final Identity identity)
+        throws IOException {
+        return response;
+    }
+
+    /**
+     * Entry interface that is used to check if the received information is
+     * valid.
+     *
+     * @author Endrigo Antonini (teamed@endrigo.com.br)
+     * @version $Id$
+     * @since 0.20
+     */
+    public interface Entry {
+
+        /**
+         * Check if is a valid user.
+         * @param user User
+         * @param pwd Password
+         * @return Identity.
+         */
+        Opt<Identity> check(String user, String pwd);
+    }
+
+    /**
+     * Entry with fixed credentials.
+     *
+     * @author Endrigo Antonini (teamed@endrigo.com.br)
+     * @version $Id$
+     * @since 0.20
+     *
+     */
+    public static final class Fixed implements PsBasic.Entry {
```

Comment:
@antonini doesn't look like a useful class for other users.. I can't imagine a situation where this class will be necessary in production code. can you?